### PR TITLE
OS specific end-line

### PR DESCRIPTION
**Pull request by fragmuffin**
Save YAML files with line-endings prescribed by the executing OS (#310).
Doesn't effect loading of yaml:

_note_
```
>>> with open('REQ001.yml', 'r') as fh:
...     content = fh.read()
... 
>>> import yaml
>>> yaml.load(content) == yaml.load(content.replace('\n', '\r\n'))
True
```

### CODE REVIEW

**Comment by fragmuffin** [comment]
Although... I tested this on a windows box... doesn't work.
I haven't used `abc` before, have I changed the wrong code?

**Comment by jacebrowning** [comment]
I believe the place to do this is not while manipulating strings (`'\n'` means newline regardless of opperating system: https://pythonconquerstheuniverse.wordpress.com/2011/05/08/newline-conversion-in-python-3/), but when writing to disk, which occurs in https://github.com/jacebrowning/doorstop/blob/develop/doorstop/common.py.

Have said that, I'm not sure the proposed change is a good idea as I think it will result in constant file changes when working on the same project accross multiple opperating systems.

**Comment by fragmuffin** [comment]
@jacebrowning 

> I'm not sure the proposed change is ideal as it will result in constant file changes when working on the same project across multiple operating systems

Yes; I'm inclined to agree.

Perhaps the solution to this problem is simply an addition to a FAQ, or advice for installation on Windows.
What do you think?

**Comment by jacebrowning** [comment]
Yeah, we could add another top-level docs page between **Home** and **Command-line Interface** (https://doorstop.readthedocs.io) called **Setup** which is an extended version of the installation steps from the README including tips for configuruing Windows.

What do you think?

File: docs/setup.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Setup
+
+Doorstop can require different settings for different systems.
+This page contains recommendations for setting up your repository to
+accomodate a variety of systems.
+
+**Intended Audience:** CVS Repository Administrator
```
[the remark this answers is not in the document]
Comment (fragmuffin) [reply docs/setup.md]: Yes, you're absolutely right

**Comment by jacebrowning** [comment]
Thanks!

**Comment by fragmuffin** [comment]
Oh!, you're too quick @jacebrowning... I was just in the process of testing the suggestion, and there's 2 problems:
1. attribute has a typo: `safeclrf` should be `safecrlf`
1. the suggestion doesn't actually work...

I managed to revert my system back to default by running:

```
:: core.safecrlf
git config --unset core.safecrlf
git config --local --unset core.safecrlf
git config --global --unset core.safecrlf
git config --system --unset core.safecrlf
:: core.autocrlf
git config --unset core.autocrlf
git config --local --unset core.autocrlf
git config --global --unset core.autocrlf
git config --system --unset core.autocrlf
```

It *should* work according to the git docs.
But applying the described "fix" doesn't remove the warnings. I tried with all combinations of:
* `.gitattributes` file in repo' root dir, and folder containing `.yml` files
* `* <attribute name> false` and `*.yml <attribute name> false`, and
* just `core.autocrlf`, just `core.safecrlf`, or both set to `false`

All attempts failed.

However, if I run in a terminal:

```
git config core.autocrlf false
```

Before running Doorstop I don't get the warnings (yes `core.autocrlf`... `core.safecrlf` to false doesn't work)

Then, that alters the repository, or the system, not just the terminal's local environment;
If I close that down, open a fresh terminal, and re-run the Doorstop command, I *still* don't get any warnings, but there's no change to any `.gitattributes` file.

**Conclusion**
I certainly don't understand how git applies its attributes (in windows anyway... the above tested on: `git version 2.17.1.windows.2`).
The only method I found that consistently works is to call `git config core.autocrlf false` before executing the Doorstop command, or in my case, a test-suite populating Doorstop "requirements" as traceable test-results.
(note: I also tried `git config --local ...` in an attempt to limit the change to the running environment, but even _that_ was persistent)

Aghh! 😠 I find Windows infuriating, but I have to use it for work.

**Proposed Change**
I change the recommended setting to to the above `git config` command, instead of a `.gitattribute`.
I don't like it, but the fact is the proposed solution simply isn't working.

**Comment by jacebrowning** [comment]
That must be changing some global Git setting on your machine.

That proposed change sounds fine to me. Please submit another PR to update the docs! ✌️

**Comment by fragmuffin** [comment]
@jacebrowning no problem, I'll do that tonight.
sorry for the run-around 👎

**Pull request merged**